.travis.yml

```diff
@@ -1,26 +1,26 @@
 language: python
 sudo: false
 python:
-  - 2.7
-  - 3.4
-  - 3.5
-  - 3.6
-  - pypy-5.4.1
+    - 2.7
+    - 3.4
+    - 3.5
+    - 3.6
+    - pypy
+    - pypy3
+env:
+    - DEP=test
+matrix:
+  include:
+    - python: 2.7
+      env: DEP=no_such_extra
+install:
+    - pip install -U pip setuptools
+    - pip install -U coverage coveralls zope.testing zope.testrunner
+    - pip install -U -e .[$DEP]
 script:
-  - coverage run -m zope.testrunner --test-path=src  --auto-color --auto-progress
-
+    - coverage run -m zope.testrunner --test-path=src
```

Comment:
*sniff* I'll miss the pretty colors

Reply:
Heh, I just assumed that they didn't actually do anything since *something* must be capturing the stdout for us to see them in the log and --auto-XXX says they only apply to a terminal. But now that I actually look, there really is colour there (well, was)! So either Travis CI is really smart about how it captures output (more than just 'cmd > file') or zope.testrunner isn't that smart about what it considers a terminal.

Either way, it works! No reason to take them out, then.

Reply:
Travis CI *is* really smart and nice and interprets ANSI colors (and, apparently, uses pseudoterminals?).